client/src/views/page-components/account/AccountHistoryModal/AccountHistoryModal.ts

```diff
@@ -0,0 +1,388 @@
+      this.setState(e.detail); // undefined ( 애초에 가져올만한 정보는 결제수단목록뿐) !! 값 확인하기 !!
+
+      this.modalWrapper = createDOMWithSelector('div', '.account-history-wrapper');
+      this.render();
+
+      const payMethodForm = $('.history-form__pay-method');
+      this.dateInput = $('.history-form__date');
+      this.moneyInput = $('.history-form__money');
+
+      this.payMethod = new PayMethods({ parent: payMethodForm, state: samplePay }); // 결제수단의 정보 갖고있어야함!
+      this.modalWrapper.addEventListener('click', this.onClickHandler.bind(this));
+      this.modalWrapper.addEventListener('keyup', this.onKeyUpHandler.bind(this));
+      this.dateInput.addEventListener('focusout', this.onFocusOutDateInputHandler.bind(this));
+      this.moneyInput.addEventListener('focusout', this.onFocusOutMoneyInputHandler.bind(this));
+      this.dateInput.addEventListener('focusin', this.onFocusInDateInputHandler.bind(this));
+      this.moneyInput.addEventListener('focusin', this.onFocusInMoneyInputHandler.bind(this));
+    });
+  }
+
+  onFocusInDateInputHandler(e: MouseEvent) {
+    const { target } = e;
+    if (!(target instanceof HTMLElement)) return;
+    this.checkInputValueOnlyNumberRegex(target);
+  }
+
+  onFocusInMoneyInputHandler(e: MouseEvent) {
+    const { target } = e;
+    if (!(target instanceof HTMLElement)) return;
+    this.checkInputValueOnlyNumberRegex(target);
+  }
+
+  onClickHandler(e: MouseEvent) {
+    this.onClickCategory(e);
+    this.onClickOverlay(e);
+    this.onClickSubmitButton(e);
+  }
+
+  onClickSubmitButton(e: MouseEvent) {
+    const { target } = e;
+    if (!(target instanceof HTMLElement)) return;
+    if (target.className === 'history-form__confirm') {
+      e.preventDefault();
+      this.checkAllValidation();
+    }
+  }
+
+  checkAllValidation() {
+    const historyContent: HTMLInputElement = document.querySelector('.history-form__content');
+
+    if (
+      this.payMethod.currentCardName.length > 0 &&
+      this.dateValueValidation &&
+      this.moneyValueValidation &&
+      this.choicedCategoryName.length > 0 &&
+      historyContent.value.length > 0
+    ) {
+      this.checkIncomeOrExpenditure();
+
+      const submitArguments = {
+        //   user:,
+        card: this.payMethod.currentCardName,
+        category: this.choicedCategoryName,
+        money: this.moneyInput.value as HTMLInputElement,
+        date: this.dateInput.value as HTMLInputElement,
+        content: historyContent.value,
+        type: this.type,
+      };
+      console.log('Form Success');
+
+      this.closeModal();
+      handleEvent.fire('createaccounthistory', { state: history.state, submitArguments });
+      // 옵저버 발동
+    } else {
+      this.showAlert('.history-form__confirm-alert');
+    }
+  }
+
+  checkIncomeOrExpenditure() {
+    if (this.choicedCategoryName === '용돈') {
+      this.type = 'income';
+    } else {
+      this.type = 'expenditure';
+      this.moneyInput.value = '-' + this.moneyInput.value;
+    }
```

Comment:
if else라면 else case에서 choicedCategoryName의 이외의 요소를 방법이 없습니다 ㅜㅜ
early return 을 한다면 조금 더 좋은 코드가 될 것 같습니다!!

```suggestion
    if (this.choicedCategoryName === '용돈') return this.type = 'income';
    
    this.type = 'expenditure';
    this.moneyInput.value = '-' + this.moneyInput.value;
```

Reply:
좋은 지적입니다 감사합니다:)

Reply:
다만 
" choicedCategoryName의 이외의 요소를 방법이 없습니다 ㅜㅜ "
이 얘기에서 early return 해도 이외 요소를 모르지않나요..?! 궁금하네요!